### ISSUE TITLE
Silent in focus push notification displayed

### ISSUE THREAD

**Issue by tapz** [OneSignal/OneSignal-Android-SDK#265]
When the app is active, a push notification with contents, data, and android_sound is not displayed, but when the push notification does not have any of those fields (= it should be silent regardless if the app is active or not) the notification is displayed in the Notification Shade. I have set inFocusDisplaying to None.

OneSignal.setInFocusDisplaying(OneSignal.OSInFocusDisplayOption.None);

Version: 3.5.8
Devices: 
Moto G 4th gen, Android 7.0
Samsung Galaxy J5 2017 (SM-J530F), Android 7.0

**Comment by jkasten2** [comment]
If the notification does not have a message body (contents) it won't be display in any case by our SDK.

If the above is correct could you attach the full OneSignal REST API payload or the exact notification options you used on the OneSignal dashboard?

**Comment by tapz** [comment]
This is what I send and it calls notificationReceived, but also displays the notification when the app is active.

POST https://onesignal.com/api/v1/notifications
{
"app_id": "00000000-0000-0000-0000-000000000000",
"priority": "10",
"android_group": "mygroup",
"content_available": "true",
"filters"=[
{
"key": "MyKey1", 
"field": "tag", 
"value": "true", 
"relation": "="
}, {
"key": "MyKey2", 
"field": "tag", 
"value": "true", 
"relation": "="
}]
}

**Comment by tapz** [comment]
Anything new about this? Easily reproducible with curl:

```
curl --include \
     --request POST \
     --header "Content-Type: application/json" \
     --header "Authorization: Basic 7b8ayb890yaep89vby8adpb89adyb89HIOH99IHiohhihihih" \
     --data-binary "{\"content_available\": \"true\", \"app_id\": \"000000000-0000-0000-0000-000000000000\", \"priority\": \"10\", \"android_sound\": \"nil\", \"android_visibility\": \"-1\", \"android_group\": \"mygroup\", \"filters\": [{\"field\": \"tag\", \"key\": \"abc\", \"relation\": \"=\", \"value\": \"true\"}, {\"field\": \"tag\", \"key\": \"efg\", \"relation\": \"=\", \"value\": \"true\"}]}" \
     https://onesignal.com/api/v1/notifications
```

**Comment by tapz** [comment]
Adding this to AndroidManifest.xml and `hidden: true` to the notification request body seems to help:

```
<service
        android:name="com.geektime.rnonesignalandroid.NotificationNotDisplayingExtender"
        android:exported="false">
        <intent-filter>
          <action android:name="com.onesignal.NotificationExtender" />
        </intent-filter>
</service>
```

**Comment by jkasten2** [comment]
@tapz Fixed this issue in commit 75e7c333bf4d3908ba6cc0381fc7e44ab157ea9a and added a test to prevent future occurrences of this.
